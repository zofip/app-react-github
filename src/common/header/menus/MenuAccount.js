import React, { useState } from "react";

import { IconButton, Menu } from "@material-ui/core";
import AccountCircle from "@material-ui/icons/AccountCircle";


export default function MenuAccount() {
  const [anchorEl, setAnchorEl] = useState(null);

  const open = Boolean(anchorEl);


  function handleMenu(event) {
    setAnchorEl(event.currentTarget);
  }

  function handleClose() {
    setAnchorEl(null);
  }

  return (
    <div>
      <IconButton
        aria-label="account of current user"
        aria-controls="menu-appbar"
        aria-haspopup="true"
        onClick={handleMenu}
        color="inherit"
      >
        <AccountCircle />
      </IconButton>
      <Menu
        id="menu-appbar"
        anchorEl={anchorEl}
        anchorOrigin={{
          vertical: "top",
          horizontal: "right"
        }}
        keepMounted
        transformOrigin={{
          vertical: "top",
          horizontal: "right"
        }}
        open={open}
        onClose={handleClose}
      >
      </Menu>
    </div>
  );
}
