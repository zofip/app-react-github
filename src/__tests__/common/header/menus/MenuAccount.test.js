import React from 'react';
import { shallow } from 'enzyme';
import MenuAccount from '../../../../common/header/menus/MenuAccount';

describe('<MenuAccount />', () => {
   it('Rendering', () => {
      const menuAccount = shallow(<MenuAccount />);
      expect(menuAccount).toMatchSnapshot();
   });
});