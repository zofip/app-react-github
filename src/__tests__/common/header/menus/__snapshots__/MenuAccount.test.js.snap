// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MenuAccount /> Rendering 1`] = `
<div>
  <WithStyles(ForwardRef(IconButton))
    aria-controls="menu-appbar"
    aria-haspopup="true"
    aria-label="account of current user"
    color="inherit"
    onClick={[Function]}
  >
    <AccountCircleIcon />
  </WithStyles(ForwardRef(IconButton))>
  <WithStyles(ForwardRef(Menu))
    anchorEl={null}
    anchorOrigin={
      Object {
        "horizontal": "right",
        "vertical": "top",
      }
    }
    id="menu-appbar"
    keepMounted={true}
    onClose={[Function]}
    open={false}
    transformOrigin={
      Object {
        "horizontal": "right",
        "vertical": "top",
      }
    }
  />
</div>
`;
